Add merge-pr command

mullr opened the pull request.

Commit:
Add git-merge-pr command

Commit:
Add basic integration test for merge-pr

Commit:
Update to latest version of git-rsl

Commit:
Fix and test --no-prune option to merge-pr

Commit:
Add test assertions for branch state

Commit:
Clean up warnings

Commit:
Clean up merge-pr error reporting

windwardrail commented on src/git-merge-pr.rs:
```diff
@@ -4,18 +4,171 @@ extern crate git_rsl;
+
+    // The original shell version runs 'git gc' here; I'm omitting it because it
+    // gc is a potentially destructive operation, and doing it automatically is
+    // somewhat dangerous.
```
This kind of documentation should probably not live in the code. If it needs to be investigated, then an issue would be good. Otherwise, it would ideally live in a design document.

windwardrail commented on src/git-merge-pr.rs:
```diff
@@ -4,18 +4,171 @@ extern crate git_rsl;
+    // that won't work here because git-rsl's implementation of secure-push can
+    // only push a branch to a remote version of the same name (i.e. you can't
+    // push HEAD:branchname, only branchname:branchname)
+    ws.repo.set_head(&local_destination_branch)?;
```
Same issue with documentation

windwardrail commented on src/git-merge-pr.rs:
```diff
@@ -4,18 +4,171 @@ extern crate git_rsl;
+                  opts.destination_branch,
+                  opts.destination_branch,
+                  opts.destination_branch)
+    }
```
You can put a number inside the braces that references the index of the argument ex. `"your local {0} branch ...`. Then you only have to list the argument once.

mullr replied:
woah, that's a cool feature.

windwardrail commented on tests/fixture/mod.rs:
```diff
@@ -0,0 +1,126 @@
+        .output()
+        .unwrap();
+
+//    "pub:u:1024:17:31BC51B4BB930CA9:1528761082:::u:::scaSCA:::::::"
```
Can this be removed?

mullr replied:
I left it in as an example of what the data looks like, so  the following parsing step makes sense. I'll amend this with more comments about what it means.

Commit:
Fix test after error reporting change

The previous commit moved error message from stderr to stdout, following the
convention of git-rsl. This updates the test code to follow.

Commit:
Dry up format block

Commit:
Provide context for the sample gpg with-colons output
